resolwe/flow/executors/__init__.py

```diff
@@ -57,43 +108,52 @@ def iterjson(text):
         yield obj
 
 
-class BaseFlowExecutor(BaseEngine):
+class BaseFlowExecutor(object):
     """Represents a workflow executor."""
 
     exported_files_mapper = defaultdict(dict)
 
     def __init__(self, *args, **kwargs):
         """Initialize attributes."""
-        super(BaseFlowExecutor, self).__init__(*args, **kwargs)
         self.data_id = None
         self.process = None
         self.requirements = {}
         self.resources = {}
-
-    def hydrate_spawned_files(self, filename, data_id):
-        """Hydrate spawned files' paths."""
-        if filename not in self.exported_files_mapper[self.data_id]:
-            raise KeyError('Use `re-export` to prepare the file for spawned process: {}'.format(filename))
-
-        export_fn = self.exported_files_mapper[self.data_id].pop(filename)
-
-        if self.exported_files_mapper[self.data_id] == {}:
-            self.exported_files_mapper.pop(self.data_id)
-
-        return {'file_temp': export_fn, 'file': filename}
+        if EXECUTING:
+            self.redis = redis.StrictRedis()
+            self.manager_cookie = 1
+            self.reply_chan = '{}.{}'.format(settings['REDIS_CHANNEL_PAIR'][1], settings['Data_get']['id'])
+
+    def make_settings(self, data_id):
+        """Add any additional required settings into what the manager will serialize."""
+        if EXECUTING:
+            raise RuntimeError('BaseFlowExecutor.make_config() called while running')
+        return {
+            'USE_TZ': settings.USE_TZ,
+            'FLOW_EXECUTOR_TOOLS_PATHS': self.get_tools(),
+            'Data_get': model_to_dict(Data.objects.get(
+                pk=data_id
+            )),
+            'Process_get': model_to_dict(Data.objects.get(
+                pk=data_id
+            ).process),
+        }
 
     def get_tools(self):
         """Get tools paths."""
-        tools_paths = []
-        for app_config in apps.get_app_configs():
-            proc_path = os.path.join(app_config.path, 'tools')
-            if os.path.isdir(proc_path):
-                tools_paths.append(proc_path)
-
-        custom_tools_paths = getattr(settings, 'RESOLWE_CUSTOM_TOOLS_PATHS', [])
-        if not isinstance(custom_tools_paths, list):
-            raise KeyError("`RESOLWE_CUSTOM_TOOLS_PATHS` setting must be a list.")
-        tools_paths.extend(custom_tools_paths)
+        if not EXECUTING:
+            tools_paths = []
+            for app_config in apps.get_app_configs():
+                proc_path = os.path.join(app_config.path, 'tools')
+                if os.path.isdir(proc_path):
+                    tools_paths.append(proc_path)
+
+            custom_tools_paths = getattr(settings, 'RESOLWE_CUSTOM_TOOLS_PATHS', [])
+            if not isinstance(custom_tools_paths, list):
+                raise KeyError("`RESOLWE_CUSTOM_TOOLS_PATHS` setting must be a list.")
+            tools_paths.extend(custom_tools_paths)
+        else:
+            tools_paths = settings.FLOW_EXECUTOR_TOOLS_PATHS
```

Comment:
Are tools copied to the executor? Or should they already be there? At the moment they are a part of python packages, i.e. are installed on local disc, usually not shared storage.

cc @tjanez

Reply:
If I'm following right, the new idea is to make a separate management command to do the copying and then modify the environment enough to support testing again. The manager wouldn't be doing the copying, since the files are expected to rarely change. Instead, django's static file collection thingy is supposed to be used.

Reply:
Yes. Copying tools to an appropriate directory is a once-per-deploy tasks and doesn't belong into the manager. I would go with something similar to Django's `collectstatic` command.

Reply:
I think @dblenkus' idea is fine for now.